src/loop_logic.rs

```diff
@@ -10,16 +10,37 @@ use std::time::{Duration, Instant};
 use std::future::Future;
 
 use io_lifetimes::AsFd;
-use slotmap::SlotMap;
+use slab::Slab;
 
 use crate::sources::{Dispatcher, EventSource, Idle, IdleDispatcher};
+use crate::sys::Notifier;
 use crate::{EventDispatcher, InsertError, Poll, PostAction, TokenFactory};
 
 type IdleCallback<'i, Data> = Rc<RefCell<dyn IdleDispatcher<Data> + 'i>>;
 
-slotmap::new_key_type! {
-    pub(crate) struct CalloopKey;
-}
+// The maximum number of sources that we are allowed to have.
```

Comment:
It's not the maximum number of sources, it's the number of bits used to store the source id.